File: fundamentals-of-programming/exam/exam-partial/src/domain/product.py
'''
Created on Dec 2, 2013

@author: daniel
'''


class Product(object):
    def __init__(self, product_id, product_name, product_price):
        self.__id = product_id
        self.__name = product_name
        self.__price = product_price

    def getID(self):
        return self.__id

    def getName(self):
        return self.__name

    def getPrice(self):
        return self.__price

    def setId(self, value):
        self.__id = value

    def setName(self, value):
        self.__name = value

    def setPrice(self, value):
        self.__price = value

    def __str__(self):
        return "\tName: %s, Price: %s" % (self.getName(), str(self.getPrice()))

File: aspect-oriented-programming/project/src/project/model/Product.java
package project.model;


public class Product
{
    private int id;
    private String name;
    private int quantity;
    private float price;

    public Product(int id, String name, int quantity, float price)
    {
        this.id = id;
        this.name = name;
        this.quantity = quantity;
        this.price = price;
    }

    public int getId()
    {
        return id;
    }

    public String getName()
    {
        return name;
    }

    public int getQuantity()
    {
        return quantity;
    }

    public float getPrice()
    {
        return price;
    }

    @Override
    public String toString()
    {
        return "Product{" +
            "id=" + id +
            ", name='" + name + '\'' +
            ", quantity=" + quantity +
            ", price=" + price +
            '}';
    }
}
